pandas/core/strings.py

```diff
@@ -742,10 +735,7 @@ def str_split(arr, pat=None, n=None, return_type='series'):
                 n = 0
             regex = re.compile(pat)
             f = lambda x: regex.split(x, maxsplit=n)
-    if return_type == 'frame':
-        res = DataFrame((Series(x) for x in _na_map(f, arr)), index=arr.index)
```

Comment:
yes, the logic is replaced to ``_wrap_results_expand``. 

I think #10081 should be considered separately, because we can't simply use workaround for all the cases.

```
s = pd.Series([1.1, '2.2'])

# current behavior
s.str.split('.', expand=True)
#      0    1
# 0  NaN  NaN
# 1    2    2

# workaround or numpy.split
pd.read_table(StringIO(s.to_csv(None, index=None)), sep='.')
#    1  1.1
# 0  2    2
```